web/ui/react-app/src/pages/Status.tsx

```diff
@@ -54,8 +55,18 @@ export const statusConfig: StatusConfig = {
   droppedAlertmanagers: { skip: true },
 };
 
-const Status = () => {
-  const { response: data, error, isLoading } = useFetches<StatusPageState[]>(ENDPOINTS);
+const endpointsMemo: { [prefix: string]: string[] } = {};
+
+const Status: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix = '' }) => {
+  if (!endpointsMemo[pathPrefix]) {
+    // TODO: REMOVE THIS SUPER UGLY HACK IMMEDIATELY.
+    // The problem is that there's an infinite reload loop if the endpoints array is
+    // reconstructed on every render, as the dependency checking in useFetches()
+    // then thinks that something has changed... the whole useFetches() should
+    // probably removed and solved differently (within the component?) somehow.
```

Comment:
Just curious on your thoughts to why it should be removed, too many small unexpected edge cases?

Reply:
I think I went overboard with my comment here. It's not that bad, but basically in the actual app (vs. tests etc.), there's only ever one path prefix throughout the lifetime of the application, and thus these endpoints are basically constants. Caching them in a global mutable object with provisions for different path prefixes seems like more than should be needed, except that constructing them each time on the fly in the render function causes infinite render loops due to  `useEffect()` array dependency comparisons to fail.